### PR TITLE
`[EXILED::API]` Adding SpawnMice

### PR DESCRIPTION
**Pull request by NotZer0Two**
## Description
**Describe the changes** 
Lets people spawn rats

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?** (if this is a feature change)

**Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

Credits to Yamato for the finding
(and the idea to add it)
<br />

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentations
<br />

## Submission checklist

- [x] I have checked the project can be compiled
- [x] I have tested my changes and it worked as expected

### Patches (if there are any changes related to Harmony patches)
- [ ] I have checked no IL patching errors in the console

### Other
- [ ] Still requires more testing

### CODE REVIEW

**Review by louis1706 — APPROVED** [review]

File: EXILED/Exiled.API/Features/Map.cs [hunk 1]
```diff
@@ -417,6 +424,19 @@ public static void PlayGunSound(Vector3 position, ItemType firearmType, byte max
...
         }
 
+        /// <summary>
+        /// Spawns a Mice inside the <see cref="RoomType.EzShelter"/>.
+        /// </summary>
+        /// <param name="mice">The type of mice you want to spawn..</param>
+        public static void SpawnMice(byte mice = 1)
+        {
+            if (mice > SqueakSpawner.mice.Length)
+                new OverflowException($"The mice type are from 1 to {SqueakSpawner.mice.Length}");
```
Comment (VALERA771) [inline EXILED/Exiled.API/Features/Map.cs:434]:
```suggestion
                new ArgumentOutOfRangeException($"The mice type are from 1 to {SqueakSpawner.mice.Length}");
```

Do you really want to use exception here? You can use just `Log.Error`. 
Also, just creating a new instance of exception won't stop execution of method

**Review by VALERA771 — CHANGES_REQUESTED** [review]

**Comment by VALERA771** [comment]
Can you create an enum with all mouse types?

**Comment by NotZer0Two** [comment]
> Can you create an enum with all mouse types?

The mouse type is just a mouse that spawn in a different position so nothing crazy and the name how should it be named like MiceOnTopOfTheBarrel etc...?

**Comment by NotZer0Two** [comment]
I don't mean that but the names for the different rats because the rats are different only by their position

**Comment by VALERA771** [comment]
Ah, okay. Then forgot about enum

**Review by Misfiy — APPROVED** [review]

**Review by VALERA771 — APPROVED** [review]

**Pull request merged**